File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.c
char* makeFancyString(char* s) {
    int n=strlen(s),c=1,k=0;
    char arr[n];
    arr[k++]=s[0];
    for(int i=0;i<n-1;i++){
        if(s[i]==s[i+1])c++;
        else c=1;
        if(c<3)arr[k++]=s[i+1];
    }
    s[k]='\0';
    for(int i=0;i<k;i++)s[i]=arr[i];
    return s;;
}

File: 1957-delete-characters-to-make-fancy-string/1957-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb=new StringBuilder();
        int c=1;
        sb.append(s.charAt(0));
        for(int i=1;i<s.length();i++){
            if(s.charAt(i-1)==s.charAt(i))c++;
            else c=1;
            if(c<=2)sb.append(s.charAt(i));
        }
        return sb.toString();
    }
}